docker-code-server-custom-image-dotfiles/main.tf

```diff
@@ -0,0 +1,114 @@
+##### Variables
+# These are custom values provided by the user
+# when creating a Workspace using this template.
+#
+# At the time of writing, there are two:
+# 1. dotfiles_uri: a dotfiles repo to clone on workspace start
+# 2. custom_docker_image: the base image for the workspace
+
+variable "dotfiles_uri" {
+  description = <<-EOF
+  Dotfiles repo URI (optional)
+
+  see https://dotfiles.github.io
+  EOF
+  default     = ""
+}
+
+variable "custom_docker_image" {
+  description = <<-EOF
+  Base image for your workspace
+
+  i.e. codercom/enterprise-base:ubuntu
+  EOF
+  default     = "codercom/enterprise-base:ubuntu"
+}
+
+##### Variables - END
+
+##### code-server
+# Things required to include code-server in the Workspace.
+resource "coder_agent" "main" {
+  arch           = "amd64"
+  os             = "linux"
+  startup_script = <<EOF
+    #!/bin/sh
+    set -x
+    # install and start code-server
+    curl -fsSL https://code-server.dev/install.sh | sh
+    code-server --auth none --port 13337 &
+    sudo service docker start
+    coder dotfiles -y ${var.dotfiles_uri}
+    EOF
+}
+
+resource "coder_app" "code-server" {
+  agent_id = coder_agent.main.id
+  name     = "code-server"
+  url      = "http://localhost:13337/?folder=/home/coder"
+  icon     = "/icon/code.svg"
+}
+
+##### code-server - END
+
+##### coder
+# Miscellaneous things required by Coder
+# to make Docker and everything work as expected.
+# Copied from https://github.com/coder/coder/blob/main/examples/templates/docker-with-dotfiles/main.tf
+terraform {
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = "0.4.9"
+    }
+    docker = {
+      source  = "kreuzwerker/docker"
+      version = "~> 2.20.2"
+    }
+  }
+}
+
+data "coder_provisioner" "me" {
+}
+
+provider "docker" {
+}
+
+data "coder_workspace" "me" {
+}
```

Comment:
One thing that was confusing to me was if I could delete these or not. It's not obvious what they're used for or why they're needed.

Reply:
Ahhh got it. Okay I'll delete those. Thanks man!